app/models/region.rb

```diff
@@ -60,6 +61,21 @@ def log_payload
 
   private
 
+  def initialize_path?
+    @initialize_path
+  end
+
+  def initialize_path
+    errors.add(:reparent_to, "must assign new parent to initialize path") unless reparent_to
+    logger.info(class: self.class, msg: "got reparent_to: #{reparent_to}, going to initialize new path")
+    self.path = "#{reparent_to.path}.#{path_label}"
+    self.reparent_to = nil
```

Comment:
Why do we need to set this back to `nil`?

Reply:
Being defensive and making sure we mark the record as "done reparenting".  There are cases where validation callbacks on a model could be triggered multiple times.